### PR TITLE
Added /favorite/leaderboard API endpoint

### PR DESCRIPTION
**Pull request by mickeyn**
This new endpoint will be used to replace the query sent from
WEB.

### CODE REVIEW

File: lib/MetaCPAN/Server/Controller/Favorite.pm [hunk 1]
```diff
@@ -45,5 +45,12 @@ sub recent : Path('recent') : Args(0) {
     $c->stash($data);
 }
 
+sub leaderboard : Path('leaderboard') : Args(0) {
+    my ( $self, $c ) = @_;
+    my $data = $self->model($c)->raw->leaderboard();
+    $data or return;
+    $c->stash($data);
```
Comment (oalders) [inline lib/MetaCPAN/Server/Controller/Favorite.pm]: In general, I wonder if we should just shorten these to:

`return $c->stash( $self->model($c)->raw->leaderboard || {} );`

Not sure if you can stash an `undef`.

**Comment by oalders** [comment]
@haarg this build appears to have failed bec. Carton is trying to get `provides` via the `v0` endpoint.

File: lib/MetaCPAN/Server/Controller/Favorite.pm (hunk 1 above)
Comment (mickeyn) [inline lib/MetaCPAN/Server/Controller/Favorite.pm]: you can't stash a non-hashref value ... that's where this check comes from.
i don't mind the suggested change, but i assigned $data for debugging purposes.

**Comment by mickeyn** [comment]
the tests passed now, but the v0 error is still thrown in the setup test

**Pull request merged**